File: ClockBellCounter-Java/src/test/java/kbra/app/ClockBellCounterTest.java
package kbra.app;

import static org.junit.Assert.assertEquals;

import java.time.LocalTime;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class ClockBellCounterTest
{
    ClockBellCounter bellCounter;
    LocalTime startTime;
    LocalTime endTime;
    Integer expectedBells;
    Integer bells;
    
    @Before
    public void init() {
        bellCounter = new ClockBellCounter();
    }

    @Test
    public void countBells_exactHoursMorning()
    {
        startTime = LocalTime.of(2, 0);
        endTime = LocalTime.of(3, 0);
        expectedBells = 5;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_exactHoursAfternoon()
    {
        startTime = LocalTime.of(14, 0);
        endTime = LocalTime.of(15, 0);
        expectedBells = 5;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_notTheExactHours()
    {
        startTime = LocalTime.of(14, 23);
        endTime = LocalTime.of(15, 42);
        expectedBells = 3;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_oneDaytoAnother()
    {
        startTime = LocalTime.of(23, 0);
        endTime = LocalTime.of(1, 0);
        expectedBells = 24;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_withinSameHour()
    {
        startTime = LocalTime.of(12, 15);
        endTime = LocalTime.of(12, 19);
        expectedBells = 0;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_sameHourDifferentDay()
    {
        startTime = LocalTime.of(16, 20);
        endTime = LocalTime.of(16, 20);
        expectedBells = 158;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
    
    @Test
    public void countBells_sameHourDifferentMinuteDifferentDay()
    {
        startTime = LocalTime.of(1, 25);
        endTime = LocalTime.of(1, 1);
        expectedBells = 156;
        bells = bellCounter.countBells(startTime.toString(), endTime.toString());
        assertEquals( expectedBells, bells );
    }
}

File: ClockBellCounter-PHP/src/test/ClockBellCounterTest.php
<?php

namespace test;

require "src/kbra/ClockBellCounter.php";

use PHPUnit\Framework\TestCase;
use kbra\ClockBellCounter;

class ClockBellCounterTest extends TestCase
{
    
    public function test_countBells_exactHoursMorning()
    {
        $clockBellCounter = new ClockBellCounter();
        $startTime = "2:00";
        $endTime = "3:00";
        $expectedBells = 5;
        
        $bells = $clockBellCounter->countBells($startTime, $endTime);
        
        $this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_exactHoursAfternoon()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "14:00";
    	$endTime = "15:00";
    	$expectedBells = 5;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_notTheExactHours()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "14:23";
    	$endTime = "15:42";
    	$expectedBells = 3;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_oneDaytoAnother()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "23:00";
    	$endTime = "1:00";
    	$expectedBells = 24;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_withinSameHour()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "12:15";
    	$endTime = "12:19";
    	$expectedBells = 0;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_sameHourDifferentDay()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "16:20";
    	$endTime = "16:20";
    	$expectedBells = 158;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
    
    public function test_countBells_sameHourDifferentMinuteDifferentDay()
    {
    	$clockBellCounter = new ClockBellCounter();
    	$startTime = "1:25";
    	$endTime = "1:01";
    	$expectedBells = 156;
    
    	$bells = $clockBellCounter->countBells($startTime, $endTime);
    
    	$this->assertEquals($expectedBells, $bells);
    }
}